src/networking.c

```diff
@@ -2292,9 +2307,14 @@ int processInlineBuffer(client *c) {
 
     /* Setup argv array on client structure */
     if (argc) {
-        if (c->argv) zfree(c->argv);
-        c->argv_len = argc;
-        c->argv = zmalloc(sizeof(robj*)*c->argv_len);
+        /* Create new argv if space is insufficient or the new arguments are too large. */
+        if (unlikely(argc > c->argv_len ||
+            (c->argv_len > ARGV_CACHE_THRESHOLD && c->argv_len > argc * 2)))
```

Comment:
so we're re-using the allocation only if it's very small, and not wasting more than half of it.
considering that it's very small, do really care how much of it is wasted?
or alternatively, maybe for big ones, if they're only 10% wasted we may want to re-use?

or maybe the benefit for saving that one allocation on large argv arrays is negligible (considering the elements we put in the array too)?

actually, now that i think of it, over a certain size we do gradual allocations 
```
        c->argv_len = min(c->multibulklen, 1024);
```
so it probably doesn't make sense to re-use in these cases (we're wasting time on reallocs anyway)

Reply:
good idea.

Reply:
i change the threshold to 1024, and free the arguments in cron.
This also means that we don't have to worry about constantly rebuilding when the command size fluctuates greatly, which is more in line with the actual usage scenario.
Because I remember some users using `ping` for keep alive, it could result in never being reused.